subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/inception/BuildSrcApiChangePerformanceTest.groovy

```diff
@@ -61,6 +59,40 @@ class BuildSrcApiChangePerformanceTest extends AbstractCrossVersionPerformanceTe
         result.assertCurrentVersionHasNotRegressed()
     }
 
+    def "buildSrc non-abi change"() {
+        given:
+        runner.tasksToRun = ['help']
+        runner.runs = determineNumberOfRuns(runner.testProject)
+
+        and:
+        runner.addBuildMutator { settings ->
+            def changingClassFilePath = new File(settings.projectDir, 'buildSrc/src/main/groovy/ChangingClass.groovy')
+            new BuildMutator() {
+                @Override
+                void beforeScenario(ScenarioContext context) {
+                    writeFile(changingClassFilePath, """
+                        class ChangingClass {
+                            void changingMethod${context.phase}${context.iteration}() {
+                                System.out.println("Do the thing");
+                            }
+                        }
+                    """)
+                }
+
+                @Override
+                void beforeBuild(BuildContext context) {
+                    writeFile(changingClassFilePath, """
+                        class ChangingClass {
+                            void changingMethod${context.phase}${context.iteration}() {
```

Comment:
Why don't you use the `ApplyNonAbiChangeToGroovySourceFileMutator` mutator? Note that currently you are doing an ABI change, since you change the method name.

Reply:
`ApplyNonAbiChangeToGroovySourceFileMutator` gave me a CNFE.

Reply:
Fixed.